Implement _pad_circular in ATen

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #73434
* #73433
* #73432
* #73431
* __->__ #73410

Closes #44459

This migrates the python implementation of `_pad_circular` to ATen and
removes the old C++ implementation that had diverged from python.

Note that `pad` can't actually use this until the
forward-compatibility period is over.

Differential Revision: [D34571403](https://our.internmc.facebook.com/intern/diff/D34571403)

peterbell10 committed:
Migrate _pad_circular to ATen

Closes #44459

This migrates the python implementation of `_pad_circular` to ATen and
removes the old C++ implementation that had diverged from python.

[ghstack-poisoned]

peterbell10 committed:
Update on "Implement _pad_circular in ATen"


Closes #44459

This migrates the python implementation of `_pad_circular` to ATen and
removes the old C++ implementation that had diverged from python.

Note that `pad` can't actually use this until the
forward-compatibility period is over.

[ghstack-poisoned]

peterbell10 committed:
Update on "Implement _pad_circular in ATen"


Closes #44459

This migrates the python implementation of `_pad_circular` to ATen and
removes the old C++ implementation that had diverged from python.

Note that `pad` can't actually use this until the
forward-compatibility period is over.

[ghstack-poisoned]

peterbell10 committed:
Update on "Implement _pad_circular in ATen"


Closes #44459

This migrates the python implementation of `_pad_circular` to ATen and
removes the old C++ implementation that had diverged from python.

Note that `pad` can't actually use this until the
forward-compatibility period is over.

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34571403).

ezyang commented:
For reference, the Python implementation lives here https://github.com/pytorch/pytorch/blob/2f957f513e6faebb001d60a544aedd8042ce7026/torch/nn/functional.py#L4722

ezyang reviewed (approved):
Cool generalization. Did you try flipping over the Python implementation to the C++ one to get better test coverage?

ezyang commented:
A comment at the condemned code pointing to this PR and saying what the BC window is would help us remember when it can be axed (and, in the meantime, where bugfixes should be applied in two places).

ezyang commented:
oops, I just realized I got codeowner tagged. @mruberry feel free to commandeer the diff stack from me

peterbell10 committed:
Add comments that python code will be removed on "Implement _pad_circular in ATen"


Closes #44459

This migrates the python implementation of `_pad_circular` to ATen and
removes the old C++ implementation that had diverged from python.

Note that `pad` can't actually use this until the
forward-compatibility period is over.

Differential Revision: [D34571403](https://our.internmc.facebook.com/intern/diff/D34571403)

[ghstack-poisoned]

ezyang commented:
Need a little more warning cleanliness for internal build

```
stderr: caffe2/aten/src/ATen/native/PadNd.cpp:90:30: error: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'long' [-Werror,-Wsign-compare]
  TORCH_CHECK(padding.size() == ndim * 2,
              ~~~~~~~~~~~~~~ ^  ~~~~~~~~
stderr: caffe2/aten/src/ATen/native/PadNd.cpp:90:30: error: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'long' [-Werror,-Wsign-compare]
  TORCH_CHECK(padding.size() == ndim * 2,
              ~~~~~~~~~~~~~~ ^  ~~~~~~~~
```

peterbell10 committed:
Update on "Implement _pad_circular in ATen"


Closes #44459

This migrates the python implementation of `_pad_circular` to ATen and
removes the old C++ implementation that had diverged from python.

Note that `pad` can't actually use this until the
forward-compatibility period is over.

Differential Revision: [D34571403](https://our.internmc.facebook.com/intern/diff/D34571403)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34571403).

peterbell10 committed:
Rebase and fix merge conflicts on "Implement _pad_circular in ATen"


Closes #44459

This migrates the python implementation of `_pad_circular` to ATen and
removes the old C++ implementation that had diverged from python.

Note that `pad` can't actually use this until the
forward-compatibility period is over.

Differential Revision: [D34571403](https://our.internmc.facebook.com/intern/diff/D34571403)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34571403).

ezyang commented:
@pytorchbot merge this
